### PR TITLE
show better error message for sevensegment demo

### LINKED ISSUE #71: sevensegment_demo.py fails
```
pi@raspberrypi:~/Git/luma.examples/examples $ python sevensegment_demo.py
Version: luma.oled 2.3.1 (luma.core 1.2.1)
Display: ssd1306
Interface: i2c
Dimensions: 128 x 64
------------------------------------------------------------
Traceback (most recent call last):
  File "sevensegment_demo.py", line 115, in <module>
    main()
  File "sevensegment_demo.py", line 71, in main
    seg = sevensegment(device)
  File "/usr/local/lib/python2.7/dist-packages/luma/core/virtual.py", line 482, in __init__
    self.segment_mapper = segment_mapper or device.segment_mapper
AttributeError: 'ssd1306' object has no attribute 'segment_mapper'

```

### PR DESCRIPTION
**Pull request by thijstriemstra**
when using unsupported devices.

fixes #71

### CODE REVIEW

**Comment by thijstriemstra** [comment]
ps. i haven't tested this on python 2.7, 3.4 only.. (rpi2 jessie)

**Comment by rm-hull** [comment]
I think it he emulators have a seven seg transform that should be accommodated as well

**Comment by thijstriemstra** [comment]
@rm-hull in that case this isn't a good fix. Ideally we should be able to query a luma library for their sevensegment devices. Thoughts?

**Comment by rm-hull** [comment]
Maybe do a `hasattr` check on the device to see if it has the `segment_mapper` defined?
Or simply do a try/catch for the specific attribute error?

**Comment by thijstriemstra** [comment]
Not that it's terribly important but this PR doesn't have any effect on OSX either;

```
$ python examples/sevensegment_demo.py 
usage: sevensegment_demo.py [-h] [--config CONFIG] [--display] [--width WIDTH]
                            [--height HEIGHT] [--rotate] [--interface]
                            [--i2c-port I2C_PORT] [--i2c-address I2C_ADDRESS]
                            [--spi-port SPI_PORT] [--spi-device SPI_DEVICE]
                            [--spi-bus-speed SPI_BUS_SPEED] [--gpio GPIO]
                            [--gpio-data-command GPIO_DATA_COMMAND]
                            [--gpio-reset GPIO_RESET]
                            [--gpio-backlight GPIO_BACKLIGHT]
                            [--block-orientation] [--mode] [--framebuffer]
                            [--bgr] [--h-offset H_OFFSET]
                            [--v-offset V_OFFSET] [--backlight-active]
                            [--transform] [--scale SCALE]
                            [--duration DURATION] [--loop LOOP]
                            [--max-frames MAX_FRAMES]
sevensegment_demo.py: error: I2C device not found: /dev/i2c-1
```

File: examples/demo_opts.py [hunk 1]
```diff
@@ -56,12 +56,12 @@ def get_device(actual_args=None):
         config = cmdline.load_config(args.config)
         args = parser.parse_args(config + actual_args)
 
+    print(display_settings(args))
```
Comment (rm-hull) [inline examples/demo_opts.py]: makes sense 👍

**Comment by rm-hull** [comment]
For OSX, I just checked, and the seven segment demo command args are:

    $ python examples/sevensegment_demo.py -d pygame  --transform=seven_segment --width=8 --height=8

That continues to work with 937eb9c that I just pushed

File: examples/sevensegment_demo.py [hunk 2]
```diff
@@ -68,6 +69,11 @@ def show_message_alt(seg, msg, delay=0.1):
 def main():
     # create seven segment device
     device = get_device()
+
+    if not hasattr(device, 'segment_mapper'):
```
Comment (rm-hull) [inline examples/sevensegment_demo.py]: Hmmm -  thinking about it, this check would be better off in the `sevensegment` class instead

**Comment by thijstriemstra** [comment]
Looks like that didn't help either, e.g.:

```
$ python examples/sevensegment_demo.py -d ssd1306
Version: luma.oled 2.3.1 (luma.core 1.2.1)
Display: ssd1306
Interface: i2c
Dimensions: 128 x 64
------------------------------------------------------------
usage: sevensegment_demo.py [-h] [--config CONFIG] [--display] [--width WIDTH]
                            [--height HEIGHT] [--rotate] [--interface]
                            [--i2c-port I2C_PORT] [--i2c-address I2C_ADDRESS]
                            [--spi-port SPI_PORT] [--spi-device SPI_DEVICE]
                            [--spi-bus-speed SPI_BUS_SPEED] [--gpio GPIO]
                            [--gpio-data-command GPIO_DATA_COMMAND]
                            [--gpio-reset GPIO_RESET]
                            [--gpio-backlight GPIO_BACKLIGHT]
                            [--block-orientation] [--mode] [--framebuffer]
                            [--bgr] [--h-offset H_OFFSET]
                            [--v-offset V_OFFSET] [--backlight-active]
sevensegment_demo.py: error: I2C device not found on address: 0x3C
```

**Comment by rm-hull** [comment]
Presumably that fails earlier in the init, because OSX doesnt have the i2c device ?

p.s. it would be useful if the version printed something like: 

```
Version: luma.oled 2.3.1 (luma.core 1.2.1) on ${platform.platform} with Python X.Y.Z
```

**Pull request merged**